MapboxMobileEventsTests/MMEExceptionalDictionary.m

```diff
@@ -0,0 +1,22 @@
+#import "MMEExceptionalDictionary.h"
+
+@implementation MMEExceptionalDictionary
+
+- (instancetype) initWithObjects:(id  _Nonnull const [])objects forKeys:(id<NSCopying>  _Nonnull const [])keys count:(NSUInteger)cnt {
+    return [super init];
+}
+
+- (instancetype) mutableCopy {
+    [[NSException exceptionWithName:NSGenericException reason:@"testing" userInfo:nil] raise];
+    return nil;
+}
+
+- (NSUInteger) count {
+    return 0;
+}
+
+- (NSEnumerator *)keyEnumerator {
```

Comment:
[NSDictionary documentation](https://developer.apple.com/documentation/foundation/nsdictionary?language=objc#1776559) states that `-objectForKey:` needs to be overridden as well.

Reply:
It does, provided you don't want the subclass to throw an exception when a caller attempts to call that method, which in this exceptional case, is the desired behavior.